express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const viewNotes = require('./viewNotes');
+const fs = require('fs');
+const data = require('./data.json');
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  res.status(200).json(viewNotes(null));
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const noteID = parseInt(req.params.id);
+  if (noteID < 0 || !Number.isInteger(noteID)) {
+    return res.status(400).json({ Error: 'ID is not a valid positive integer' });
+  } else {
+    const singleNote = viewNotes(noteID);
+
+    if (singleNote) {
+      res.status(200).json(singleNote);
+    } else {
+      res.status(404).json({ Error: 'Selected ID does not exist in database' });
+    }
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  const requestNote = req.body;
+  if (requestNote.content === undefined) {
+    return res.status(400).json({ Error: 'No content included in request body' });
+  } else {
+    const dataObj = data;
+    requestNote.id = dataObj.nextId;
+    dataObj.notes[dataObj.nextId] = requestNote;
+    dataObj.nextId++;
+    const updatedData = JSON.stringify(dataObj, null, 2);
+    res.status(201).json(requestNote);
+    fs.writeFile('data.json', updatedData, 'utf8', err => {
+      if (err) {
+        console.error(err);
+        res.status(500).json({ Error: 'An unexpected error occured' });
+      }
+    });
```

Comment:
Please don't send a successful response until you're sure the change was saved to the file successfully.